components/patient-data/rest/src/main/java/org/phenotips/data/rest/internal/DefaultPatientsSuggestionsResourceImpl.java

```diff
@@ -0,0 +1,266 @@
+            safeOrder = " desc";
+        }
+
+        StringBuilder querySb = new StringBuilder();
+        querySb.append("select doc.name from  Document doc, doc.object(PhenoTips.PatientClass) as patient");
+        querySb.append(" where doc.name <> :t and lower(doc.name) like :").append(INPUT_PARAMETER);
+        querySb.append(" or lower(patient.external_id) like :").append(INPUT_PARAMETER);
+
+        boolean usePatientName = this.configuration.getConfiguration("patient").getEnabledFieldNames()
+            .contains(FIRST_NAME);
+        if (usePatientName) {
+            querySb.append(" or lower(patient.first_name) like :").append(INPUT_PARAMETER);
+            querySb.append(" or lower(patient.last_name) like :").append(INPUT_PARAMETER);
+        }
+
+        querySb.append(" order by " + safeOrderField + safeOrder);
+        try {
+            Query query = this.qm.createQuery(querySb.toString(), Query.XWQL);
+            query.setLimit(maxResults);
+            query.bindValue("t", "PatientTemplate");
+            String formattedInput = String.format(INPUT_FORMAT, input);
+            query.bindValue(INPUT_PARAMETER, formattedInput);
+
+            queryResults = query.execute();
+        } catch (QueryException e) {
+            this.logger.error("Error while performing patiets query: [{}] ", e.getMessage());
+            return Collections.emptyList();
+        }
+        return queryResults;
+    }
+
+    private JSONObject getPatientJSON(Patient patient, boolean markFamilyAssociation)
+    {
+        JSONObject patientJSON = new JSONObject();
+        patientJSON.put("id", patient.getId());
+        patientJSON.put("identifier", patient.getExternalId());
+
+        // Add patient URL
+        patientJSON.put("url", getURL(patient));
+
+        String description = getDescription(patient);
+
+        // Add family info
+        Family family = this.familyRepository.getFamilyForPatient(patient);
+        if (family != null) {
+            patientJSON.put("familyId", family.getId());
+            if (markFamilyAssociation) {
+                description.concat(", in family: ").concat(family.getId());
+            }
+        }
+
+        // Add description
+        patientJSON.put("textSummary", description);
+
+        return patientJSON;
+    }
+
+    private void appentPatientXmlResult(Patient patient, StringBuilder xmlResult, boolean markFamilyAssociation)
+    {
+        String escapedReference = XMLUtils.escapeXMLComment(patient.getDocumentReference().toString());
+
+        xmlResult.append("<rs id=\"").append(getURL(patient)).append("\" ");
+        xmlResult.append("info=\"").append(escapedReference).append("\">");
+
+        String description = getDescription(patient);
+
+        // Add family info
+        Family family = this.familyRepository.getFamilyForPatient(patient);
+        if (family != null) {
+            xmlResult.append("familyId=\"").append(family.getId()).append("\" ");
```

Comment:
This shows up in the label, since the tag was closed already on line 221.

Reply:
Seems we did not use `familyId` anyways (based on [old code](https://github.com/phenotips/phenotips/pull/2467/files#diff-5422a2cd9f7ae082de4ecf1ee418661dL136) ) so I'll just remove it in next commit